source/Body.cpp

```diff
@@ -205,90 +251,399 @@ const Government *Body::GetGovernment() const
+	};
+
+	// Default to flying state sprite.
+	Body::BodyState state = Body::BodyState::FLYING;
+	auto pair = SPRITE_LOAD_PARAMS.find(node.Token(0));
+	if(pair != SPRITE_LOAD_PARAMS.end())
+		state = pair->second;
+
+	const Sprite *sprite = SpriteSet::Get(node.Token(1));
+	SpriteParameters *spriteData = &sprites[state];
+	SpriteParameters::AnimationParameters spriteAnimationParameters;
+	std::vector<DataNode> triggerSpriteDefer;
 
 	// The only time the animation does not start on a specific frame is if no
 	// start frame is specified and it repeats. Since a frame that does not
 	// start at zero starts when the game started, it does not make sense for it
 	// to do that unless it is repeating endlessly.
 	for(const DataNode &child : node)
 	{
-		if(child.Token(0) == "frame rate" && child.Size() >= 2 && child.Value(1) >= 0.)
-			frameRate = child.Value(1) / 60.;
+		if(child.Token(0) == "trigger" && child.Size() >= 2)
+		{
+			// Defer loading any trigger sprites until after the main sprite is loaded.
+			// Ensures that all "default" parameters are loaded first.
+			triggerSpriteDefer.push_back(child);
+		}
+		else if(child.Token(0) == "center" && child.Size() >= 3)
+			spriteAnimationParameters.center = Point(child.Value(1), child.Value(2));
+		else if(child.Token(0) == "frame rate" && child.Size() >= 2 && child.Value(1) >= 0.)
+			spriteAnimationParameters.frameRate = child.Value(1) / 60.;
 		else if(child.Token(0) == "frame time" && child.Size() >= 2 && child.Value(1) > 0.)
-			frameRate = 1. / child.Value(1);
+			spriteAnimationParameters.frameRate = 1. / child.Value(1);
 		else if(child.Token(0) == "delay" && child.Size() >= 2 && child.Value(1) > 0.)
-			delay = child.Value(1);
+			spriteAnimationParameters.delay = child.Value(1);
+		else if(child.Token(0) == "transition delay" && child.Size() >= 2 && child.Value(1) >= 0.)
+			spriteAnimationParameters.transitionDelay = child.Value(1);
 		else if(child.Token(0) == "scale" && child.Size() >= 2 && child.Value(1) > 0.)
-			scale = static_cast<float>(child.Value(1));
+			spriteAnimationParameters.scale = static_cast<float>(child.Value(1));
 		else if(child.Token(0) == "start frame" && child.Size() >= 2)
 		{
-			frameOffset += static_cast<float>(child.Value(1));
-			startAtZero = true;
+			spriteAnimationParameters.startFrame = static_cast<float>(child.Value(1));
+			spriteAnimationParameters.startAtZero = true;
+		}
+		else if(child.Token(0) == "ramp" && child.Size() >= 2)
+		{
+			if(child.Size() == 2)
+			{
+				spriteAnimationParameters.rampUpRate = child.Value(1) / 3600.;
+				spriteAnimationParameters.rampDownRate = child.Value(1) / 3600.;
+			}
+			else if(child.Size() == 3)
+			{
+				spriteAnimationParameters.rampUpRate = child.Value(1) / 3600.;
+				spriteAnimationParameters.rampDownRate = child.Value(2) / 3600.;
+			}
+		}
+		else if(child.Token(0) == "indicate percentage" && child.Size() >= 2 && child.Value(1) > 0. && child.Value(1) < 1.)
```

Comment:
Percentage is a value between 0 and 100. Internally you can store it as a [0,1] float as long as it is well documented, but if you have an attribute called percentage, it either should not be called as such, or should require a real percentage.

Reply:
Agree with this, I will go ahead and divide the value by `100.` therefore expecting an actual percent from the sprite definition instead of a float.